### PR TITLE
Added option for user to choose the infra and node network interface …

### PR DESCRIPTION
**Pull request by satishoc**

### CODE REVIEW

File: update_ign.py [hunk 1]
```diff
@@ -38,201 +38,379 @@
...
+    if ans=="1":
+        choice = 1
+    elif ans=="2":
+        choice = 2
+    elif ans == "3":
+        choice = 3
+    elif ans == "4":
+        choice = 4
+    else:
+        print("\n   Invalid Option, Please Try Again")
```
Comment (snaiksat) [inline update_ign.py:317]: Could you please provide information about these options as a part of the comments section we have at the top of this script? We intend that to be the documentation on how to run this script and what options it supports.

Comment (satishoc) [reply update_ign.py:317]: Added

File: update_ign.py [hunk 2]
```diff
@@ -37,202 +49,365 @@
...
+            BONDING_MASTER=yes
+            BONDING_OPTS="mode=4 miimon=100 lacp_rate=1"
+            NAME=""" + bond_name + """
+            TYPE=Bond
+            MTU=""" + mtu + """
+            """).encode()
+        return ifcfg_bond0
+
+
+    def create_interface_with_bond(self, interface_name, bond_name, mtu):
```
Comment (abhis2112) [inline update_ign.py]: create_slave_interface

File: update_ign.py [hunk 3]
```diff
@@ -37,202 +49,365 @@
...
+            infra_network_interface1_b64 = base64.standard_b64encode(infra_network_interface1).decode().strip()
+            config_data[interface_name] = {'base64': infra_network_interface1_b64, 'path': '/etc/sysconfig/network-scripts/'  + interface_name}
+             
+            interface_name = "ifcfg-" + self.aci_infra_network_interface[1]
+
+            infra_network_interface2 = self.create_interface_with_bond(self.aci_infra_network_interface[1], "bond1", self.opflex_network_mtu)
+            infra_network_interface2_b64 = base64.standard_b64encode(infra_network_interface2).decode().strip()
+            config_data[interface_name] = {'base64': infra_network_interface2_b64, 'path': '/etc/sysconfig/network-scripts/' + interface_name}
+
+            opflex_conn = ("""VLAN=yes
```
Comment (abhis2112) [inline update_ign.py]: this code is repeated. Please modularize it

File: update_ign.py [hunk 4]
```diff
@@ -37,202 +49,365 @@
...
+            config_data['ifcfg_opflex_conn'] = {'base64': ifcfg_opflex_conn_b64, 'path': '/etc/sysconfig/network-scripts/ifcfg-opflex-conn'}
+
+            route_opflex_conn = """ADDRESS0=224.0.0.0
+        NETMASK0=240.0.0.0
+        METRIC0=1000
+        """.encode()
+
+            route_opflex_conn_b64 = base64.standard_b64encode(route_opflex_conn).decode().strip()
+
+            config_data['route_opflex_conn'] = {'base64': route_opflex_conn_b64, 'path': '/etc/sysconfig/network-scripts/route-opflex-conn'}
```
Comment (abhis2112) [inline update_ign.py]: till here. Make a separate function if this code will be called again and again

File: update_ign.py [hunk 5]
```diff
@@ -37,202 +49,365 @@
...
+                'contents': {
+                    'source': 'data:text/plain;charset=utf-8;base64,' + hostname_b64,
+                    'verification': {}
+                },
+                'filesystem': 'root',
+            })
+
+        if 'bootstrap' not in hostname.decode():
+
+            for k, v in config_data.items():
```
Comment (abhis2112) [inline update_ign.py]: make it more verbose. Like - interface, interface_config --> interface_config[path]

Comment (satishoc) [reply update_ign.py]: done

File: update_ign.py (hunk 2 above)
Comment (satishoc) [reply update_ign.py]: renamed

File: update_ign.py (hunk 4 above)
Comment (satishoc) [reply update_ign.py]: done

File: update_ign.py [hunk 6]
```diff
@@ -21,7 +21,19 @@
...
 
-# How to run this script: python3 update_ign.py
+#               we will have following option to choose from 
+#                    1.Single interface for node network and bond interface for infra network
+#                    2.Bond interface for both node and infra networks
+#                    3.Bond interface for node network and single interface for infra network
+#                    4.Single interface for both node and infra networks
+#
+#
+#               currently we support option one only(please update interface name in config.yaml file)
```
Comment (snaiksat) [inline update_ign.py]: > currently we support option one only

This comment is fine in the commit message but does not make sense here.  

> (please update interface name in config.yaml file)

This should be removed from the parentheses and clearly stated what changes need to be made in each case. So say something to the effect of:

    Please update the config.yaml as follows:

File: update_ign.py [hunk 7]
```diff
@@ -21,7 +21,19 @@
...
+
+
+# How to run this script: 
+#           Please update the config.yaml file before running this script
```
Comment (snaiksat) [inline update_ign.py]: Please remove this comment

File: update_ign.py [hunk 8]
```diff
@@ -21,7 +21,19 @@
...
+#           Trigger the command after changing config.yaml file according to requirement:   python3 update_ign.py
```
Comment (snaiksat) [inline update_ign.py]: Please revert this back to how it was earlier

File: update_ign.py [hunk 9]
```diff
@@ -21,7 +21,19 @@
 #                     has four or more interfaces then you can choose to have bonded pairs for both the node and infra networks.
 #                     In either of those cases this script will have to be modified to account for those differences.
 
-# How to run this script: python3 update_ign.py
+#               we will have following option to choose from
```
Comment (snaiksat) [inline update_ign.py]: How does this sentence tie into the existing text comments above it??

**Review by snaiksat — CHANGES_REQUESTED** [review]

**Comment by snaiksat** [comment]
Thanks @satishoc for updating the comment. @abhis2112  when you get a chance please check if the generated output provided by @satishoc for all the cases matches the expectations.

**Review by abhis2112 — APPROVED** [review]

**Comment by abhis2112** [comment]
lgtm

**Review by snaiksat — APPROVED** [review]

**Pull request merged**